lib/write.js

```diff
@@ -3,12 +3,18 @@
 var mqtt = require('mqtt-packet')
 
 function write (client, packet, done) {
-  var result = mqtt.writeToStream(packet, client.conn)
-
-  if (!result && !client.errored && done) {
-    client.conn.once('drain', done)
-  } else if (done) {
-    setImmediate(done)
+  if (client.conn.writable && client.connected) {
+    var result = mqtt.writeToStream(packet, client.conn)
+    if (!result && !client.errored && done) {
+      console.log('drain')
```

Comment:
I guess this is spurious

Reply:
you mean `if (client.conn.writable && client.connected) {`
or `if (!result && !client.errored && done) {` ?

Reply:
I mean the console.log.

Reply:
ar. good catch. This is my debug purpose only. I will remove it. :)